pkg/controller/kubefed/kubefed_controller.go

```diff
@@ -150,22 +150,24 @@ func resourceEnvUpdate(scope kubefedv1alpha1.InstallationScope, ns, name string)
 		case "deployment":
 			if containers, ok, err := unstructured.NestedSlice(u.Object,
 				"spec", "template", "spec", "containers"); ok {
-				if envs, envOk, envErr := unstructured.NestedSlice(containers[0].(map[string]interface{}), "env"); envOk {
-					if !checkEnvExists(envs, "name", "DEFAULT_KUBEFED_SCOPE") {
-						fse := map[string]interface{}{"name": "DEFAULT_KUBEFED_SCOPE", "value": kubefedScope}
-						envs = append(envs, fse)
+				envs, envOk, envErr := unstructured.NestedSlice(containers[0].(map[string]interface{}), "env")
+				if !envOk {
+					if err := unstructured.SetNestedSlice(containers[0].(map[string]interface{}), envs, "env"); err != nil {
+						reqLogger.Info("Failed to set the env variable field")
 					}
```

Comment:
else should have a log statement that says, you created the `env` slice for debuggability.

Reply:
Fixed